.github/workflows/release.yml

```diff
@@ -0,0 +1,115 @@
+  rustToolchain: nightly-2019-11-04
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    strategy:
+      fail-fast: true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1.0.6
+        with:
+          toolchain: ${{ env.rustToolchain }}
+          override: true
+
+      - name: Install wasm-pack
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: wasm-pack --version 0.8.1
+
+      - name: Install Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.15.0'
+
+      - name: Build
+        run: node ./run dist --skip-version-validation --target linux
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: ide-dist
+          path: dist/client/linux-unpacked
+
+
+  upload-cdn:
+    name: Prepare Release
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+
+      - name: Install Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.15.0'
+
+      - name: Prepare Distribution Version
+        shell: bash
+        run: |
+          ref=${{ github.ref }}
+          refversion=${ref#"refs/tags/ide-"}
+          echo "DIST_VERSION=$refversion" >> $GITHUB_ENV
+
+      - name: Extract ASAR file
+        shell: bash
+        run: |
+          npx asar extract ./artifacts/ide-dist/resources/app.asar ./dist
+
+      - name: Prepare AWS Session
+        shell: bash
+        run: |
+          aws configure --profile s3-upload <<-EOF > /dev/null 2>&1
+          ${{ secrets.ARTEFACT_S3_ACCESS_KEY_ID }}
+          ${{ secrets.ARTEFACT_S3_SECRET_ACCESS_KEY }}
+          us-west-1
+          text
+          EOF
+
+      - name: Upload the index.js to S3
+        shell: bash
+        run: >
+          aws s3 cp ./dist/assets/index.js.gz
+          s3://ensocdn/ide/${{ env.DIST_VERSION }}/index.js.gz --profile
+          s3-upload --acl public-read --content-encoding gzip
+
+      - name: Upload the ide.wasm to S3
+        shell: bash
+        run: >
+          aws s3 cp ./dist/assets/ide.wasm
+          s3://ensocdn/ide/${{ env.DIST_VERSION }}/ide.wasm --profile
+          s3-upload --acl public-read --content-type 'application/wasm'
+
+      - name: Upload the wasm_imports.js to S3
+        shell: bash
+        run: >
+          aws s3 cp ./dist/assets/wasm_imports.js.gz
+          s3://ensocdn/ide/${{ env.DIST_VERSION }}/wasm_imports.js.gz --profile
+          s3-upload --acl public-read --content-encoding gzip
+
+      - name: Teardown AWS Session
+        shell: bash
+        run: |
+          aws configure --profile s3-upload <<-EOF > /dev/null 2>&1
+          null
+          null
+          null
+          text
+          EOF
```

Comment:
oh God, what are these nulls? :O

Reply:
empty lines

Reply:
I copied it from the engine

Reply:
Removed.
Ara added this as a good practice to limit the scopes of privileged access as much as possible. I think it is unnecessary. The container has those credentials, so tearing down the AWS session doesn't increase security.